### PR TITLE
OCLOMRS-355: Implement adding bulk concepts using IDs populated in the text area

### PR DESCRIPTION
**Pull request by ozezzy**
# JIRA TICKET NAME:
[Implement adding bulk concepts using IDs populated in the text area on bulk add concepts page](https://issues.openmrs.org/browse/OCLOMRS-355)

# Summary:
The text area should allow taking in of a unique concept ID or group of comma separated IDs and actually adding them to the particular dictionary

### CODE REVIEW

File: src/components/bulkConcepts/addBulkConcepts.jsx [hunk 1]
```diff
@@ -27,15 +33,49 @@ export class AddBulkConcepts extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      cielConcepts: [],
-      conceptLimit: 10,
+      conceptIds: '',
+      redirect: false,
     };
+    this.source = 'ciel';
```
Comment (collinewait) [inline src/components/bulkConcepts/addBulkConcepts.jsx]: Is this(`this.source = 'ciel';`)  necessary? I am not observing its usage.

File: src/components/bulkConcepts/addBulkConcepts.jsx [hunk 2]
```diff
@@ -27,15 +33,49 @@ export class AddBulkConcepts extends Component {
...
+      match: {
+        params: { type, typeName, collectionName },
+      },
+    } = this.props;
+    const url = `${type}/${typeName}/collections/${collectionName}/references/`;
+    const { conceptIds } = this.state;
+    const conceptIdList = conceptIds.match(/\d+/g);
+
+    if (conceptIdList !== null) {
+      const concepts = conceptIdList.map(_ => `/orgs/CIEL/sources/CIEL/concepts/${_}/`);
```
Comment (collinewait) [inline src/components/bulkConcepts/addBulkConcepts.jsx]: Is it possible to add concepts from other sources other than `CIEL`?

File: src/components/bulkConcepts/addBulkConcepts.jsx [hunk 3]
```diff
@@ -49,19 +56,58 @@ export class AddBulkConcepts extends Component {
...
     const {
       match: {
         params: { type, typeName, collectionName },
       },
       cielConcepts,
       language,
     } = this.props;
-    const disableButton = (conceptIds.length < 1);
+
+    const disableButton = false; //(conceptIds.length < 1);
```
[the remark this answers is not in the document]
Comment (ozezzy) [reply src/components/bulkConcepts/addBulkConcepts.jsx]: Not at all. I used that for debugging, thanks for noticing it.

File: src/components/bulkConcepts/addBulkConcepts.jsx [hunk 4]
```diff
@@ -49,19 +56,58 @@ export class AddBulkConcepts extends Component {
...
+      },
+    } = this.props;
+    const url = `${type}/${typeName}/collections/${collectionName}/references/`;
+    const { conceptIds } = this.state;
+    const conceptIdList = conceptIds.split(/[\s,\r\n]+/);
+    if (conceptIdList[0]) {
+      this.invalidConceptIds = [];
+
+      const validConcepts = await conceptIdList.reduce(async (accumulator, id) => {
+        const validity = await isConceptValid({ url: `orgs/CIEL/sources/CIEL/concepts/${id}/` });
```
[the remark this answers is not in the document]
Comment (ozezzy) [reply src/components/bulkConcepts/addBulkConcepts.jsx]: Yes. At the moment CIEL is the only implemented source

[the remark this answers is not in the document]
Comment (dkayiwa) [reply src/components/bulkConcepts/addBulkConcepts.jsx]: What shall we do when we need to deal with other sources?

[the remark this answers is not in the document]
Comment (ozezzy) [reply src/components/bulkConcepts/addBulkConcepts.jsx]: A variable can be declared to hold the the source url to be changed based on the selected source and the reducer function on line 71 modified to take also the source url as parameter. so the implementation can handle multiple sources with modification

[the remark this answers is not in the document]
Comment (ozezzy) [reply src/components/bulkConcepts/addBulkConcepts.jsx]: I have refactored this, so adding more sources will only require adding a source select-change function to change the `this.sourceUrl` value appropriately.

**Pull request merged**